src/position-manager/AllowancePositionManager.sol

```diff
@@ -0,0 +1,174 @@
+    private _withdrawAllowances;
+
+  /// @notice Mapping of credit delegations.
+  mapping(address owner => mapping(address spender => mapping(uint256 reserveId => uint256 amount)))
+    private _creditDelegations;
+
+  /// @dev Constructor.
+  /// @param spoke_ The address of the spoke contract.
+  constructor(address spoke_) PositionManagerBase(spoke_) {}
+
+  /// @inheritdoc IAllowancePositionManager
+  function approveWithdraw(address spender, uint256 reserveId, uint256 amount) external {
+    _withdrawAllowances[msg.sender][spender][reserveId] = amount;
+    emit WithdrawApproval(msg.sender, spender, reserveId, amount);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function approveWithdrawWithSig(
+    EIP712Types.WithdrawPermit calldata params,
+    bytes calldata signature
+  ) external {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    address user = params.owner;
+    bytes32 digest = _hashTypedData(params.hash());
+    require(SignatureChecker.isValidSignatureNow(user, digest, signature), InvalidSignature());
+    _useCheckedNonce(user, params.nonce);
+
+    _withdrawAllowances[user][params.spender][params.reserveId] = params.amount;
+    emit WithdrawApproval(user, params.spender, params.reserveId, params.amount);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function approveCreditDelegation(address spender, uint256 reserveId, uint256 amount) external {
+    _creditDelegations[msg.sender][spender][reserveId] = amount;
+    emit CreditDelegation(msg.sender, spender, reserveId, amount);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function approveCreditDelegationWithSig(
+    EIP712Types.CreditDelegation calldata params,
+    bytes calldata signature
+  ) external {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    address user = params.owner;
+    bytes32 digest = _hashTypedData(params.hash());
+    require(SignatureChecker.isValidSignatureNow(user, digest, signature), InvalidSignature());
+    _useCheckedNonce(user, params.nonce);
+
+    _creditDelegations[user][params.spender][params.reserveId] = params.amount;
+    emit CreditDelegation(user, params.spender, params.reserveId, params.amount);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function renounceWithdrawAllowance(address owner, uint256 reserveId) external {
+    _withdrawAllowances[owner][msg.sender][reserveId] = 0;
+    emit WithdrawApproval(owner, msg.sender, reserveId, 0);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function renounceCreditDelegation(address owner, uint256 reserveId) external {
+    _creditDelegations[owner][msg.sender][reserveId] = 0;
+    emit CreditDelegation(owner, msg.sender, reserveId, 0);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function withdrawOnBehalfOf(
+    uint256 reserveId,
+    uint256 amount,
+    address onBehalfOf
+  ) external returns (uint256, uint256) {
+    require(amount > 0, InvalidAmount());
```

Comment:
spoke checks so we can avoid?

Reply:
added it as early return, but could be discarded yeah